### DISCUSSION TITLE
Which Javascript library / framework would you like to use the design system with?

### DISCUSSION THREAD

**Discussion by moiz-conceptlz** [TDRA-ae/aegov-dls#40] [Polls]
Building websites in today's modern world, would point in the direction that the CMS (a.k.a the back-end) should be separate from the front-end. Moreover, the UAE Design System is a tool not only for websites, but an ever evolving set of components and blocks that can be used to build applications as well.

Using a **Headless CMS** would most likely need the website to use a JavaScript library / framework. Although not a requirement, it is always suggested that websites must be built with a layer that allows content to be separated from presentation. 

JavaScript frameworks and JavaScript libraries are both pieces of code authored by others to solve everyday problems. One is not superior to the other, nor can one be used in place of the other. However, a developer can choose which one to utilize based on the project's requirements.

Considering that you would want to use the UAE Design System with a JavaScript library / framework, what would you vote for?

**Comment by moiz-conceptlz** [comment]
Please leave a comment if you have selected "Others" in the choices of the poll.

**Comment by almasry2011** [comment]
Hello! I just wanted to share that AngularJS & ReactJS and Vue.js are JavaScript libraries not frameworks, please add Angular+2 in vote

**Reply by moiz-conceptlz** [reply]
Thats's a good point!

**Reply by moiz-conceptlz** [reply]
Do you mean, you dont want to include AngularJS @almasry2011 ? because "AngularJS support has officially ended as of January 2022." - as per https://angularjs.org/

**Reply by almasry2011** [reply]
Angular +2 and later versions are the official Angular framework, 
and the latest version is Angular 17. 
please check  ==>  https://angular.io/

**Reply by moiz-conceptlz** [reply]
@almasry2011 got it, As we cannot change the values of the poll once created here, I suggest you select "others" and we will count your vote for Angular.io

**Comment by usmanehsan0613** [comment]
Hi there, 

as per the documentation , it says by default the javascript libraries are included in " if the bundled JavaScript library of AEGov-Design-System is utilized." 

where it does gets installed , i have installed the three plugins but modals and accordian doesnot work. I want to use exactly what TDRA has provided rather writing my own addOns

kindly advise.
